src/components/DatePicker/DatePicker.scss

```diff
@@ -169,3 +186,75 @@ $in-range-border-color: #9ca6de;
   padding-bottom: spacing(extra-tight);
   text-align: center;
 }
+
+.newDesignLanguage {
+  .Day {
+    box-shadow: none;
+
+    &:hover {
+      background: var(--p-interactive-hovered);
+      color: var(--p-text-on-interactive);
+    }
+
+    @include focus-ring;
+    // stylelint-disable-next-line selector-max-specificity
+    &:focus:not(:active) {
+      @include focus-ring($style: 'focused');
+      box-shadow: none;
+    }
+  }
+
+  .Day-selected {
+    background: var(--p-interactive);
+    border: none;
+    color: var(--p-text-on-interactive);
+  }
+
+  // stylelint-disable-next-line selector-max-specificity
+  .Day:not(.Day-lastInRange):not(.Day-firstInRange):not(.Day-inRange) {
+    border-radius: spacing(extra-tight);
+  }
+
+  .Day-firstInRange {
+    @include radius-with-focus-ring {
+      border-radius: spacing(extra-tight);
+    }
+
+    &.Day-hasRange,
+    &.Day-hoverRight {
+      @include radius-with-focus-ring {
+        @include left-half-circle;
+      }
+    }
+  }
+
+  .Day-lastInRange {
+    @include radius-with-focus-ring {
+      @include right-half-circle;
+    }
+  }
```

Comment:
```suggestion
 .Day-firstInRange {
    border-radius: var(--p-border-radius-base);

    &.Day-hasRange,
    &.Day-hoverRight {
      border-radius: 50% 0 0 50%;

      // stylelint-disable-next-line selector-max-specificity, selector-max-class
      &::after {
        border-radius: 50% 0 0 50%;
      }
    }
  }

  .Day-lastInRange {
    border-radius: 0 50% 50% 0;
    &::after {
      border-radius: 0 50% 50% 0;
    }
  }
```

Reply:
If I 🎩 correctly, this solves all the issues with the animating ring. I also find it a bit easier to follow than the mixin.